relayer-cli/src/commands/query/channel_ends.rs

```diff
@@ -0,0 +1,182 @@
+
+    let connection_id = channel_end
+        .connection_hops
+        .first()
+        .ok_or_else(|| format!("missing connection_hops"))?
+        .clone();
+
+    let connection_end = chain.query_connection(&connection_id, chain_height)?;
+
+    let client_id = connection_end.client_id().clone();
+
+    let client_state = chain.query_client_state(&client_id, chain_height)?;
+
+    let channel_counterparty = channel_end.counterparty().clone();
+    let connection_counterparty = connection_end.counterparty().clone();
+
+    let counterparty_client_id = connection_counterparty.client_id().clone();
+
+    let counterparty_connection_id = connection_counterparty.connection_id.ok_or_else(|| {
+        format!(
+            "connection end do not have counterparty connection id: {:?}",
+            connection_end
+        )
+    })?;
+
+    let counterparty_port_id = channel_counterparty.port_id().clone();
+
+    let counterparty_channel_id = channel_counterparty.channel_id.ok_or_else(|| {
+        format!(
+            "channel end do not have counterparty channel id: {:?}",
+            channel_end
+        )
+    })?;
+
+    let counterparty_chain_id = client_state.chain_id.clone();
+
+    let chain_config_b = config.find_chain(&counterparty_chain_id).ok_or_else(|| {
+        format!(
+            "chain '{}' not found in configuration file",
+            counterparty_chain_id
+        )
+    })?;
+
+    let counterparty_chain = CosmosSdkChain::bootstrap(chain_config_b.clone(), rt).unwrap();
+
+    let counterparty_chain_height = counterparty_chain.query_latest_height()?;
+
+    let counterparty_connection_end = counterparty_chain
+        .query_connection(&counterparty_connection_id, counterparty_chain_height)?;
+
+    let counterparty_client_state = counterparty_chain
+        .query_client_state(&counterparty_client_id, counterparty_chain_height)?;
+
+    let counterparty_channel_end = counterparty_chain.query_channel(
+        &counterparty_port_id,
+        &counterparty_channel_id,
+        counterparty_chain_height,
+    )?;
+
+    if cmd.summary {
+        let res = TraceSummary {
+            chain_id,
+            client_id,
+            connection_id,
+            channel_id,
+            port_id,
+
+            counterparty_chain_id,
+            counterparty_client_id,
+            counterparty_connection_id,
+            counterparty_channel_id,
+            counterparty_port_id,
+        };
+
+        Output::success(res).exit();
+    } else {
+        let res = TraceResult {
+            channel_end,
+            connection_end,
+            client_state,
+
+            counterparty_channel_end,
+            counterparty_connection_end,
+            counterparty_client_state,
+        };
+
+        Output::success(res).exit();
+    }
+
+    Ok(())
+}
+
+impl Runnable for QueryChannelEndsCmd {
+    fn run(&self) {
+        match do_run(self) {
+            Ok(()) => {}
```

Comment:
I like your approach to call in the `Ok(out)` branch the `Output::success(out)` (started to use the same in some [PRs](https://github.com/informalsystems/ibc-rs/pull/1122/commits/f97a5ba342f90b7701dd4c050fa4f8d3c6717a15)) and was wondering why did you choose to omit this approach in this case?

Reply:
The command returns two different structs depending on the summary flag. Wrapping them inside another enum will cause the auto-derived `Serialize` implementation to output different json. So I thought just outputting the result directly is simpler in this case.

Reply:
Got it, I didn't notice that!